### PR TITLE
Add cluster-proxy in chart

### PR DESCRIPTION
**Pull request by xuezhaojun**
Signed-off-by: xuezhaojun <zxue@redhat.com>

### CODE REVIEW

File: stable/cluster-proxy/templates/clusterrole.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+rules:
+  - apiGroups:
+      - cluster.open-cluster-management.io
+    resources:
+      - managedclusters
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups: ["addon.open-cluster-management.io"]
```
Comment (xuezhaojun) [inline stable/cluster-proxy/templates/clusterrole.yaml:14]: @skeeey 
Here I add the permission but still the error occurs.

Comment (skeeey) [reply stable/cluster-proxy/templates/clusterrole.yaml:14]: wait... why we need `clustermanagementaddons`? I think the error is for `clustermanagementaddons`, not for `clustermanagementaddons `, and we may only need to operate `managedclusteraddon`, right?

File: stable/cluster-proxy-addon/templates/clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  # Allow cluster-proxy-addon hub controller to get managed cluster image registries
+  - apiGroups:
+      - "imageregistry.open-cluster-management.io"
+    resources:
+      - "managedclusterimageregistries"
+      - "managedclusterimageregistries"
+    verbs:
+      - "get"
+      - "list"
+      - "watch"
```
Comment (qiujian16) [inline stable/cluster-proxy-addon/templates/clusterrole.yaml:153]: eof

File: stable/cluster-proxy-addon/templates/manager-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      containers:
+        - name: manager
+          image: {{ .Values.global.imageOverrides.cluster_proxy }}
+          imagePullPolicy: {{ .Values.imagePullPolicy }}
+          command:
+            - /manager
+          args:
+            - --leader-elect=true
+            - --signer-secret-namespace={{ .Release.Namespace }}
+            - --agent-image-name={{ .Values.global.imageOverrides.cluster_proxy }}
```
Comment (qiujian16) [inline stable/cluster-proxy-addon/templates/manager-deployment.yaml:26]: add nodeSelector and toleration here

Comment (qiujian16) [reply stable/cluster-proxy-addon/templates/manager-deployment.yaml:26]: also security context.

File: stable/cluster-proxy-addon/values.yaml [hunk 4]
```diff
@@ -24,4 +37,4 @@ anp_route:
   name: cluster-proxy-anp
 
 user_route:
-  name: cluster-proxy-user
+  name: cluster-proxy-user
```
Comment (qiujian16) [inline stable/cluster-proxy-addon/values.yaml:40]: eof

File: stable/cluster-proxy-addon/templates/manager-deployment.yaml (hunk 3 above)
Comment (xuezhaojun) [reply stable/cluster-proxy-addon/templates/manager-deployment.yaml:90]: Fixed.

File: stable/cluster-proxy-addon/values.yaml (hunk 4 above)
Comment (xuezhaojun) [reply stable/cluster-proxy-addon/values.yaml]: Fixed.

**Pull request merged**